components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,933 @@
+  subroutine eam_pio_createHeader(File)
+
+    type(file_desc_t), intent(in) :: File             ! Pio file Handle
+    integer :: retval
+
+    ! TODO change options below to match specific simulation case
+    retval=pio_put_att (File, PIO_GLOBAL, 'source', 'SCREAM')
+    retval=pio_put_att (File, PIO_GLOBAL, 'case', 'TEST 1')
+    retval=pio_put_att (File, PIO_GLOBAL, 'title', 'SCORPIO TEST')
+    retval=pio_put_att (File, PIO_GLOBAL, 'logname','THE GIT LOG HASH')
+    retval=pio_put_att (File, PIO_GLOBAL, 'host', 'THE HOST')
+    retval=pio_put_att (File, PIO_GLOBAL, 'Version', &
+           '0')
+    retval=pio_put_att (File, PIO_GLOBAL, 'revision_Id', &
+           'None')
+    retval=pio_put_att (File, PIO_GLOBAL, 'initial_file', 'NONE FOR NOW')
+    retval=pio_put_att (File, PIO_GLOBAL, 'topography_file', 'NONE FOR NOW')
+    
+  end subroutine eam_pio_createHeader
+!=====================================================================!
+  subroutine eam_init_pio_subsystem(mpicom,atm_id)
+    use shr_pio_mod,   only: shr_pio_getiosys, shr_pio_getiotype
+    
+    integer, intent(in) :: mpicom
+    integer, intent(in) :: atm_id
+
+    integer :: ierr
+
+    pio_mpicom = mpicom
+
+    call MPI_Comm_rank(pio_mpicom, pio_myrank, ierr)
+    call MPI_Comm_size(pio_mpicom, pio_ntasks , ierr)
+    
+    pio_subsystem  => shr_pio_getiosys(atm_id)
+    pio_iotype     = shr_pio_getiotype(atm_id)
+    pio_rearranger = shr_pio_getrearranger(atm_id)
+
+  end subroutine eam_init_pio_subsystem
+!=====================================================================!
+  subroutine eam_pio_createfile(File,fname)
+
+    type(file_desc_t), intent(inout) :: File             ! Pio file Handle
+    character(len=*),  intent(in)    :: fname
+    !--
+    integer :: retval
+    integer                                   :: mode
+
+    mode = pio_clobber ! Set to CLOBBER for now, TODO: fix to allow for optional mode type like in CAM
+    retval = pio_createfile(pio_subsystem,File,pio_iotype,fname,mode) 
+
+  end subroutine eam_pio_createfile
+!=====================================================================!
+  subroutine eam_pio_finalize()
+    ! May not be needed, possibly handled by PIO directly.
+
+    integer :: ierr
+
+    call PIO_finalize(pio_subsystem, ierr)
+
+  end subroutine eam_pio_finalize
+!=====================================================================!
+  subroutine errorHandle(errMsg, retVal)
+
+    implicit none
+
+    character(len=*),  intent(in)    :: errMsg
+    integer,           intent(in)    :: retVal
+
+    type(pio_file_list), pointer :: curr => NULL()
+
+    if (retVal .ne. PIO_NOERR) then
+      write(*,*) retVal,errMsg
+      ! Close all the PIO Files before aborting run
+      curr => pio_file_list_top
+      do while (associated(curr))
+        if (associated(curr%pio_file)) call PIO_closefile(curr%pio_file%pioFileDesc)
+        curr => curr%next
+      end do
+      ! Kill run
+      call mpi_abort(pio_mpicom,0,retVal)
```

Comment:
We have to be careful in scream, cause we also have Kokkos to finalize. We should use `scream_finalize_session()` before aborting. That function is currently not linked with C linker, so we also need to put
```
#ifdef __CPLUSPLUS
extern "C" {
#endif
...
#ifdef __CPLUSPLUS
}
#endif
```
in that file.

Reply:
Ahhh yes, definitely.  So I should replace `mpi_abort` in the above with `scream_finalize_session`?  Which file needs the code you put in the comment?

Reply:
No, that function only finalizes kokkos. We could, arguably, let that function also finalize MPI, but that's another discussion. What you need to do is to call `scream_finalize_session` _before_ calling `mpi_abort`. However, I think you also need to modify `scream_session.cpp`, to make those routines linked with C semantic, so that fortran can call them (that's what the ifdef shenanigans that I wrote are for).

Reply:
Is there a Fortran equivalent to "call scream_finalize_session".  I'm guessing this is a C++ routine.

Reply:
It seems like scream_finalize_session should also call mpi_abort??? Though maybe Aaron's right that this is all a red herring for F90 code???

Reply:
That is an option. After all, once kokkos has been finalized, we cannot re-initialize it, so we may as well terminate. However, other parts of the code may need some sort of clean up, so it might be best to keep the scream function limited to kokkos, and make it yield back to the calling context, which will then decide what to do.

Reply:
Hmm. Ok, I see why you may want scream_finalize and mpi_abort separated.